Firestore/Source/Remote/FSTDatastore.mm

```diff
@@ -310,25 +317,30 @@ - (void)lookupDocuments:(const std::vector<DocumentKey> &)keys
 
 - (void)invokeRPCWithFactory:(GRPCProtoCall * (^)(void))rpcFactory
                 errorHandler:(FSTVoidErrorBlock)errorHandler {
-  // TODO(mikelehen): We should force a refresh if the previous RPC failed due to an expired token,
-  // but I'm not sure how to detect that right now. http://b/32762461
-  _credentials->GetToken(
-      /*force_refresh=*/false, [self, rpcFactory, errorHandler](util::StatusOr<Token> result) {
-        [self.workerDispatchQueue dispatchAsyncAllowingSameQueue:^{
-          if (!result.ok()) {
-            errorHandler(util::MakeNSError(result.status()));
-          } else {
-            GRPCProtoCall *rpc = rpcFactory();
-            const Token &token = result.ValueOrDie();
-            [FSTDatastore
-                prepareHeadersForRPC:rpc
-                          databaseID:&self.databaseInfo->database_id()
-                               token:(token.user().is_authenticated() ? token.token()
-                                                                      : absl::string_view())];
-            [rpc start];
-          }
-        }];
-      });
+  _credentials->GetToken(_forceTokenRefresh, [self, rpcFactory,
```

Comment:
If it makes things cleaner we could extend the CredentialsProvider interface with a .InvalidateToken() method or something rather than having to pass forceTokenRefresh into GetToken().

Reply:
Done.

I thought about that, but decided not to change the interface for the sake of this bug fix. However, I think it does make things cleaner.